yarn update fails with "cannot find module 'inquirer'"

## Expected Behavior

To update to latest

## Current Behavior

Fails

## Failure Information (for bugs)

```
➜  ui git:(angular7) yarn update
yarn run v1.15.2
$ ng update @nrwl/schematics
    Updating package.json with dependency jasmine-marbles @ "0.4.0" (was "0.3.1")...
    Updating package.json with dependency rxjs @ "6.5.2" (was "6.2.2")...
    Updating package.json with dependency @nrwl/schematics @ "8.0.0" (was "6.2.0")...
    Updating package.json with dependency @nrwl/nx @ "7.8.6" (was "6.2.0")...
UPDATE package.json (3169 bytes)
warning " > bootstrap@4.3.1" has unmet peer dependency "jquery@1.9.1 - 3".
warning " > bootstrap@4.3.1" has unmet peer dependency "popper.js@^1.14.7".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "@angular/common@^7.2.0".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "@angular/core@^7.2.0".
Cannot find module 'inquirer'
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

package.json

```
{
  "name": "ui",
  "version": "3.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "./node_modules/.bin/nx lint && ng lint",
    "e2e": "ng e2e",
    "affected:apps": "./node_modules/.bin/nx affected:apps",
    "affected:build": "./node_modules/.bin/nx affected:build",
    "affected:e2e": "./node_modules/.bin/nx affected:e2e",
    "affected:test": "./node_modules/.bin/nx affected:test",
    "affected:lint": "./node_modules/.bin/nx affected:lint",
    "affected:dep-graph": "./node_modules/.bin/nx affected:dep-graph",
    "format": "./node_modules/.bin/nx format:write",
    "format:write": "./node_modules/.bin/nx format:write",
    "format:check": "./node_modules/.bin/nx format:check",
    "update": "ng update @nrwl/schematics",
    "update:check": "ng update",
    "workspace-schematic": "./node_modules/.bin/nx workspace-schematic",
    "dep-graph": "./node_modules/.bin/nx dep-graph",
    "help": "./node_modules/.bin/nx help"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "6.1.3",
    "@angular/common": "6.1.3",
    "@angular/compiler": "6.1.3",
    "@angular/core": "6.1.3",
    "@angular/forms": "6.1.3",
    "@angular/platform-browser": "6.1.3",
    "@angular/platform-browser-dynamic": "6.1.3",
    "@angular/router": "6.1.3",
    "@microsoft/office-js": "1.1.10",
    "@ngrx/effects": "6.1.0",
    "@ngrx/router-store": "6.1.0",
    "@ngrx/store": "6.1.0",
    "@ngrx/store-devtools": "6.1.0",
    "@nomadreservations/ngx-codemirror": "2.0.0",
    "@nrwl/nx": "7.8.6",
    "angular2-toaster": "6.1.0",
    "angularx-logger": "3.0.0",
    "bootstrap": "4.3.1",
    "bootstrap-sass": "^3.3.7",
    "chart.js": "2.8.0",
    "class-transformer": "0.1.9",
    "classlist.js": "1.1.20150312",
    "codemirror": "5.39.2",
    "core-js": "2.5.7",
    "moment": "2.22.2",
    "ng2-charts": "2.2.3",
    "ng2-file-upload": "1.3.0",
    "ngrx-forms": "3.0.2",
    "ngrx-store-freeze": "0.2.4",
    "ngrx-store-localstorage": "5.0.1",
    "ngx-bootstrap": "3.0.1",
    "ngx-order-pipe": "2.0.1",
    "oidc-client": "1.5.2",
    "rx-polling": "1.0.1",
    "rxjs": "6.5.2",
    "web-animations-js": "2.3.1",
    "zone.js": "0.8.26"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "0.7.4",
    "@angular/cli": "6.1.4",
    "@angular/compiler-cli": "6.1.3",
    "@angular/language-service": "6.1.3",
    "@angularclass/hmr": "2.1.3",
    "@ngrx/schematics": "6.1.0",
    "@nrwl/schematics": "8.0.0",
    "@types/codemirror": "0.0.59",
    "@types/jasmine": "2.8.8",
    "@types/jasminewd2": "2.0.3",
    "@types/node": "~8.9.4",
    "@types/office-js": "0.0.129",
    "codelyzer": "4.4.4",
    "jasmine-core": "3.1.0",
    "jasmine-marbles": "0.4.0",
    "jasmine-spec-reporter": "4.2.1",
    "karma": "2.0.5",
    "karma-chrome-launcher": "2.2.0",
    "karma-coverage-istanbul-reporter": "2.0.1",
    "karma-jasmine": "1.1.2",
    "karma-jasmine-html-reporter": "1.2.0",
    "prettier": "1.14.2",
    "protractor": "5.4.0",
    "rxjs-tslint": "0.1.5",
    "ts-node": "7.0.1",
    "tslint": "5.11.0",
    "typescript": "2.9.2"
  }
}
```

nx.json
```
{
  "npmScope": "ui",
  "implicitDependencies": {
    "angular.json": "*",
    "package.json": "*",
    "tsconfig.json": "*",
    "tslint.json": "*",
    "nx.json": "*"
  },
  "projects": {
    "cm": {
      "tags": []
    },
    "outlook": {
      "tags": []
    }
  }
}
```

angular.json
```
{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "",
  "projects": {
    "cm": {
      "root": "apps/cm/",
      "sourceRoot": "apps/cm/src",
      "projectType": "application",
      "prefix": "apps/cm",
      "schematics": {
        "@schematics/angular:component": {
          "styleext": "scss",
          "spec": false
        },
        "@schematics/angular:class": {
          "spec": false
        },
        "@schematics/angular:directive": {
          "spec": false
        },
        "@schematics/angular:guard": {
          "spec": false
        },
        "@schematics/angular:module": {
          "spec": false
        },
        "@schematics/angular:pipe": {
          "spec": false
        },
        "@schematics/angular:service": {
          "spec": false
        }
      },
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "../VQPortal/bin/Debug/netcoreapp2.1/static/",
            "deployUrl": "/static/",
            "extractCss": true,
            "index": "./apps/cm/src/index.html",
            "main": "./apps/cm/src/main.ts",
            "polyfills": "./apps/cm/src/polyfills.ts",
            "tsConfig": "./apps/cm/tsconfig.app.json",
            "assets": [
              "./apps/cm/src/callback.html",
              "./apps/cm/src/silent-refresh.html",
              "./apps/cm/src/assets",
              {
                "glob": "oidc-client.min.js",
                "input": "./node_modules/oidc-client/dist/",
                "output": "./assets/js/"
              }
            ],
            "styles": [
              "./apps/cm/src/assets/scss/build.scss"
            ],
            "stylePreprocessorOptions": {
              "includePaths": [
                "./node_modules/bootstrap-sass/assets/stylesheets/bootstrap",
                "./apps/cm/src/assets/scss"
              ]
            },
            "scripts": [
              "./apps/cm/src/assets/js/jquery-2.2.4.min.js",
              "./node_modules/chart.js/dist/Chart.min.js",
              "./apps/cm/src/assets/js/workarounds.js"
            ]
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "./apps/cm/src/environments/environment.ts",
                  "with": "./apps/cm/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "cm:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "cm:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "cm:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "main": "./apps/cm/src/test.ts",
            "polyfills": "./apps/cm/src/polyfills.ts",
            "tsConfig": "./apps/cm/tsconfig.spec.json",
            "karmaConfig": "./apps/cm/karma.conf.js",
            "styles": [
              "./apps/cm/src/assets/scss/build.scss"
            ],
            "scripts": [],
            "assets": [
              "./apps/cm/src/favicon.ico",
              "./apps/cm/src/assets"
            ]
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "./apps/cm/tsconfig.app.json",
              "./apps/cm/tsconfig.spec.json"
            ],
            "exclude": [
              "**/node_modules/**"
            ]
          }
        }
      }
    },
    "outlook": {
      "root": "apps/outlook/",
      "sourceRoot": "apps/outlook/src",
      "projectType": "application",
      "prefix": "apps/outlook",
      "schematics": {
        "@schematics/angular:component": {
          "styleext": "scss",
          "spec": false
        },
        "@schematics/angular:class": {
          "spec": false
        },
        "@schematics/angular:directive": {
          "spec": false
        },
        "@schematics/angular:guard": {
          "spec": false
        },
        "@schematics/angular:module": {
          "spec": false
        },
        "@schematics/angular:pipe": {
          "spec": false
        },
        "@schematics/angular:service": {
          "spec": false
        }
      },
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "../VQPortal/bin/Debug/netcoreapp2.1/static/outlook/",
            "deployUrl": "/static/outlook/",
            "extractCss": true,
            "index": "./apps/outlook/src/index.html",
            "main": "./apps/outlook/src/main.ts",
            "polyfills": "./apps/outlook/src/polyfills.ts",
            "tsConfig": "./apps/outlook/tsconfig.app.json",
            "assets": [
              "./apps/outlook/src/callback.html",
              "./apps/outlook/src/silent-refresh.html",
              "./apps/outlook/src/favicon.png",
              "./apps/outlook/src/assets",
              {
                "glob": "oidc-client.min.js",
                "input": "./node_modules/oidc-client/dist/",
                "output": "./assets/js/"
              },
              {
                "glob": "**/!(*.debug.*|word*|project*|powerpoint*|onenote*|excel*|access*)",
                "input": "./node_modules/@microsoft/office-js/dist/",
                "output": "./assets/js/office"
              }
            ],
            "styles": [
              "./apps/outlook/src/assets/scss/styles.scss"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "./apps/outlook/src/environments/environment.ts",
                  "with": "./apps/outlook/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "outlook:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "outlook:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "outlook:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "main": "./apps/outlook/src/test.ts",
            "polyfills": "./apps/outlook/src/polyfills.ts",
            "tsConfig": "./apps/outlook/tsconfig.spec.json",
            "karmaConfig": "./apps/outlook/karma.conf.js",
            "styles": [
              "./apps/outlook/src/styles.scss"
            ],
            "scripts": [],
            "assets": [
              "./apps/outlook/src/favicon.ico",
              "./apps/outlook/src/assets"
            ]
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "./apps/outlook/tsconfig.app.json",
              "./apps/outlook/tsconfig.spec.json"
            ],
            "exclude": [
              "**/node_modules/**"
            ]
          }
        }
      }
    }
  },
  "cli": {
    "warnings": {
      "typescriptMismatch": false,
      "versionMismatch": false
    },
    "defaultCollection": "@nrwl/schematics",
    "packageManager": "yarn"
  },
  "defaultProject": "cm"
}
```

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. run `yarn update`

### Context

Please provide any relevant information about your setup:

- version of Nx used `see package.json`
- version of Angular CLI used `8.0.0`
- `angular.json` configuration `see above`
- version of Angular DevKit used `see above`
- 3rd-party libraries and their versions `see above`
- and most importantly - a use-case that fails `yarn update`

**A minimal reproduce scenario using allows us to quickly confirm a bug (or point out coding problem) as well as confirm that we are fixing the right problem.**

### Failure Logs

Please include any relevant log snippets or files here.

## Other

Any other relevant information that will help us help you.

I was able to go from 6.2.0 to 6.4.0 using altered versions of your commands.

Then going from 6.4.0 to 7.8.6 I get different errors again, I'll look at trying to correct them

```
➜  ui git:(angular7) ✗ yarn update --from 6.4.0 --to 7.8.6 --migrate-only
yarn run v1.15.2
$ ng update @nrwl/schematics --from 6.4.0 --to 7.8.6 --migrate-only
                                                  Package "@ngrx/router-store" has an incompatible peer dependency to "@angular/common" (requires "^6.0.0" (extended), would install "8.0.0").
                                                  Package "codelyzer" has an incompatible peer dependency to "@angular/compiler" (requires ">=2.3.1 <7.0.0 || >6.0.0-beta <7.0.0" (extended), would install "8.0.0").
                                                  Package "codelyzer" has an incompatible peer dependency to "@angular/core" (requires ">=2.3.1 <7.0.0 || >6.0.0-beta <7.0.0" (extended), would install "8.0.0").
                                                  Package "@ngrx/router-store" has an incompatible peer dependency to "@angular/router" (requires "^6.0.0" (extended), would install "8.0.0").
                        Updating package.json with dependency @angular/compiler-cli @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency @angular/language-service @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency typescript @ "3.4.5" (was "2.9.2")...
                        Updating package.json with dependency @angular/animations @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency @angular/common @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency @angular/compiler @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency @angular/core @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency @angular/forms @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency @angular/platform-browser @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency @angular/platform-browser-dynamic @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency @angular/router @ "8.0.0" (was "6.1.3")...
                        Updating package.json with dependency rxjs @ "6.5.2" (was "6.2.2")...
                        Updating package.json with dependency zone.js @ "0.9.1" (was "0.8.26")...
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        packageGroup metadata of package @angular/cli is malformed.
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        Updating package.json with dependency @angular/cli @ "8.0.1" (was "6.2.4")...
            Could not find ./karma.conf.js
            It is recommended that your karma configuration sets autoWatch: true
            Prettier has been updated to 1.15.1 which has support for formatting html!
            Formatting of your code might change as you are working on each file.

            Optional: You may want to run "npm run format" as part of this update to reformat all files in your workspace.

            You can also opt out of formatting in files by adding them to the .prettierignore file in the root of your workspace.
            Typescript has been set to compile with es2015 modules
            With this update, we are changing the structure of the tsconfig files.
            A tsconfig.json has been added to all project roots which is used by editors to provide intellisense.
            The tsconfig.(app|lib|spec|e2e).json files now all extend off of the tsconfig.json in the project root.
            To find out more, visit our wiki: https://github.com/nrwl/nx/wiki/Workspace-Organization#tsconfigs
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        packageGroup metadata of package @angular/cli is malformed.
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        Updating package.json with dependency @angular/cli @ "8.0.1" (was "6.2.4")...
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        packageGroup metadata of package @angular/cli is malformed.
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        Updating package.json with dependency @angular/cli @ "8.0.1" (was "6.2.4")...
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        packageGroup metadata of package @angular/cli is malformed.
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        packageGroup metadata of package @angular/cli is malformed. Ignoring.
                        Updating package.json with dependency @angular/cli @ "8.0.1" (was "6.2.4")...
            Prettier has been updated to 1.16.4 which has a lot of small improvements
            Formatting of your code might change as you are working on each file.
            Prettier will now format *.less files you can disable this by adding '*.less' to your .prettierignore

            Optional: You may want to run "npm run format" as part of this update to reformat all files in your workspace.

            You can also opt out of formatting in files by adding them to the .prettierignore file in the root of your workspace.
CREATE .prettierignore (57 bytes)
CREATE apps/cm/tsconfig.json (100 bytes)
CREATE apps/outlook/tsconfig.json (113 bytes)
UPDATE package.json (3289 bytes)
UPDATE tsconfig.json (532 bytes)
UPDATE apps/cm/tsconfig.app.json (426 bytes)
UPDATE apps/cm/tsconfig.spec.json (339 bytes)
UPDATE apps/outlook/tsconfig.app.json (416 bytes)
UPDATE apps/outlook/tsconfig.spec.json (271 bytes)
UPDATE .vscode/extensions.json (115 bytes)
UPDATE angular.json (8750 bytes)
warning package.json: "dependencies" has dependency "@ngrx/store-devtools" with range "6.1.0" that collides with a dependency in "devDependencies" of the same name with version "7.2.0"
warning ui@3.0.0: "dependencies" has dependency "@ngrx/store-devtools" with range "6.1.0" that collides with a dependency in "devDependencies" of the same name with version "7.2.0"
warning " > @ngrx/effects@7.2.0" has incorrect peer dependency "@angular/core@^7.0.0".
warning " > @ngrx/router-store@7.2.0" has incorrect peer dependency "@angular/common@^7.0.0".
warning " > @ngrx/router-store@7.2.0" has incorrect peer dependency "@angular/core@^7.0.0".
warning " > @ngrx/router-store@7.2.0" has incorrect peer dependency "@angular/router@^7.0.0".
warning " > @ngrx/store@7.2.0" has incorrect peer dependency "@angular/core@^7.0.0".
warning " > @ngrx/store-devtools@6.1.0" has incorrect peer dependency "@ngrx/store@6.1.0".
warning " > @nrwl/nx@6.2.0" has incorrect peer dependency "jasmine-marbles@0.3.1".
warning " > angular2-toaster@6.1.0" has incorrect peer dependency "@angular/common@^6.0.0".
warning " > angular2-toaster@6.1.0" has incorrect peer dependency "@angular/compiler@^6.0.0".
warning " > angular2-toaster@6.1.0" has incorrect peer dependency "@angular/core@^6.0.0".
warning " > bootstrap@4.3.1" has unmet peer dependency "jquery@1.9.1 - 3".
warning " > bootstrap@4.3.1" has unmet peer dependency "popper.js@^1.14.7".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "@angular/common@^7.2.0".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "@angular/core@^7.2.0".
warning " > ngrx-store-localstorage@5.0.1" has incorrect peer dependency "@ngrx/store@^4.0.0 || ^5.0.0 || ^6.0.0".
warning "@angular-devkit/build-angular > @ngtools/webpack@7.3.9" has incorrect peer dependency "@angular/compiler-cli@>=5.0.0 <8.0.0 || ^7.0.0-beta.0".
warning " > @angular/compiler-cli@8.0.0" has incorrect peer dependency "typescript@>=3.4 <3.5".
warning " > codelyzer@4.5.0" has incorrect peer dependency "@angular/compiler@>=2.3.1 <8.0.0 || >7.0.0-beta <8.0.0".
            ------ Static Query Migration ------
            With Angular version 8, developers need to
            explicitly specify the timing of ViewChild and
            ContentChild queries. Read more about this here:
            https://v8.angular.io/guide/static-query-migration

            TypeScript project "apps/cm/tsconfig.app.json" has syntactical errors which could cause an incomplete migration. Please fix the following failures and rerun the migration:
            node_modules/typescript/lib/lib.es2017.d.ts(21,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(22,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(23,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(24,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(25,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(26,1): error TS1084: Invalid 'reference' directive syntax.

            Migration can be rerun with: "ng update @angular/core --from 7 --to 8 --migrate-only"


            TypeScript project "apps/outlook/tsconfig.app.json" has syntactical errors which could cause an incomplete migration. Please fix the following failures and rerun the migration:
            node_modules/typescript/lib/lib.es2017.d.ts(21,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(22,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(23,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(24,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(25,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(26,1): error TS1084: Invalid 'reference' directive syntax.

            Migration can be rerun with: "ng update @angular/core --from 7 --to 8 --migrate-only"


            The template migration strategy uses the Angular compiler internally and therefore projects that no longer build successfully after the update cannot use the template migration strategy. Please ensure there are no AOT compilation errors.

            Could not setup migration strategy for "apps/cm/tsconfig.app.json". The following error has been reported:

            Error: The Angular Compiler requires TypeScript >=3.4.0 and <3.5.0 but 2.9.2 was found instead.

            Migration can be rerun with: "ng update @angular/core --from 7 --to 8 --migrate-only"


            The template migration strategy uses the Angular compiler internally and therefore projects that no longer build successfully after the update cannot use the template migration strategy. Please ensure there are no AOT compilation errors.

            Could not setup migration strategy for "apps/outlook/tsconfig.app.json". The following error has been reported:

            Error: The Angular Compiler requires TypeScript >=3.4.0 and <3.5.0 but 2.9.2 was found instead.

            Migration can be rerun with: "ng update @angular/core --from 7 --to 8 --migrate-only"


            TypeScript project "apps/cm/tsconfig.spec.json" has syntactical errors which could cause an incomplete migration. Please fix the following failures and rerun the migration:
            node_modules/typescript/lib/lib.es2017.d.ts(21,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(22,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(23,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(24,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(25,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(26,1): error TS1084: Invalid 'reference' directive syntax.

            Migration can be rerun with: "ng update @angular/core --from 7 --to 8 --migrate-only"


            TypeScript project "apps/outlook/tsconfig.spec.json" has syntactical errors which could cause an incomplete migration. Please fix the following failures and rerun the migration:
            node_modules/typescript/lib/lib.es2017.d.ts(21,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(22,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(23,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(24,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(25,1): error TS1084: Invalid 'reference' directive syntax.
            node_modules/typescript/lib/lib.es2017.d.ts(26,1): error TS1084: Invalid 'reference' directive syntax.

            Migration can be rerun with: "ng update @angular/core --from 7 --to 8 --migrate-only"

            ------------------------------------------------
EISDIR: illegal operation on a directory, read
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Still hitting the same issue?

```
➜  ui git:(angular7) rm -rf node_modules && yarn && ng update @nrwl/schematics --force
yarn install v1.16.0
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > bootstrap@4.3.1" has unmet peer dependency "jquery@1.9.1 - 3".
warning " > bootstrap@4.3.1" has unmet peer dependency "popper.js@^1.14.7".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "@angular/common@^7.2.0".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "@angular/core@^7.2.0".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "rxjs@^6.3.3".
[4/4] 🔨  Building fresh packages...
✨  Done in 10.56s.
    Updating package.json with dependency @nrwl/schematics @ "8.1.0" (was "6.4.1")...
UPDATE package.json (3213 bytes)
warning " > bootstrap@4.3.1" has unmet peer dependency "jquery@1.9.1 - 3".
warning " > bootstrap@4.3.1" has unmet peer dependency "popper.js@^1.14.7".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "@angular/common@^7.2.0".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "@angular/core@^7.2.0".
warning " > ng2-charts@2.2.3" has incorrect peer dependency "rxjs@^6.3.3".
warning "@nrwl/schematics > @nrwl/angular > jasmine-marbles@0.5.0" has incorrect peer dependency "rxjs@^6.4.0".
Cannot find module 'inquirer'
```

Hey, I'm a co-worker of @philjones88 , I hit this too but also had a require stack printed out incase that helps

```Cannot find module 'inquirer'
Cannot find module 'inquirer'
Require stack:
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@nrwl/workspace/src/command-line/workspace-schematic.js
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@nrwl/workspace/src/command-line/nx-commands.js
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@nrwl/workspace/index.js
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@nrwl/schematics/migrations/update-6-0-0/update-6-0-0.js
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@angular-devkit/schematics/tools/export-ref.js
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@angular-devkit/schematics/tools/node-module-engine-host.js
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@angular-devkit/schematics/tools/workflow/node-workflow.js
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@angular-devkit/schematics/tools/index.js
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@angular/cli/commands/add.js
- /home/james/git/github.com/VQComms/VQ/src/ui/node_modules/@angular/cli/lib/cli/index.js
- /home/james/.config/yarn/global/node_modules/@angular/cli/lib/init.js
- /home/james/.config/yarn/global/node_modules/@angular/cli/bin/ng

```

--- a/packages/schematics/migrations/update-7-0-0/update-7-0-0.ts
+++ b/packages/schematics/migrations/update-7-0-0/update-7-0-0.ts
@@ -1,5 +1,6 @@
-import { Rule, externalSchematic, chain } from '@angular-devkit/schematics';
-import { updateJsonInTree } from '@nrwl/schematics/src/utils/ast-utils';
+import { Rule, chain } from '@angular-devkit/schematics';
+import { updateJsonInTree } from '../../src/utils/ast-utils';
+import { addUpdateTask } from '../../src/utils/update-task';
 
 export default function(): Rule {
   return chain([
@@ -17,17 +18,7 @@ export default function(): Rule {
 
       return json;
     }),
-    externalSchematic('@schematics/update', 'update', {
-      packages: ['@angular/core'],
-      from: '6.1.0',
-      to: '7.0.0',
-      force: true
-    }),
-    externalSchematic('@schematics/update', 'update', {
-      packages: ['@angular/cli'],
-      from: '6.2.0',
-      to: '7.0.1',
-      force: true
-    })
+    addUpdateTask('@angular/core', '7.0.0'),
+    addUpdateTask('@angular/cli', '7.0.1')
   ]);
 }

--- a/packages/schematics/migrations/update-7-2-0/update-7-2-0.ts
+++ b/packages/schematics/migrations/update-7-2-0/update-7-2-0.ts
@@ -14,6 +14,7 @@ import { updateJsonInTree, readJsonInTree } from '../../src/utils/ast-utils';
 import { getWorkspacePath } from '../../src/utils/cli-config-utils';
 import { offsetFromRoot } from '../../src/utils/common';
 import { stripIndents } from '@angular-devkit/core/src/utils/literals';
+import { addUpdateTask } from '../../src/utils/update-task';
 
 function getBuilders(project: any): string[] {
   return Array.from(
@@ -213,12 +214,7 @@ function switchToEs2015(host: Tree, context: SchematicContext) {
   });
 }
 
-const updateAngularCLI = externalSchematic('@schematics/update', 'update', {
-  packages: ['@angular/cli'],
-  from: '7.0.1',
-  to: '7.1.0',
-  force: true
-});
+const updateAngularCLI = addUpdateTask('@angular/cli', '7.1.0');
 
 export default function(): Rule {
   return chain([

--- a/packages/schematics/migrations/update-7-5-0/update-7-5-0.ts
+++ b/packages/schematics/migrations/update-7-5-0/update-7-5-0.ts
@@ -1,13 +1,9 @@
-import { Rule, chain, externalSchematic } from '@angular-devkit/schematics';
+import { Rule, chain } from '@angular-devkit/schematics';
 
 import { updateJsonInTree } from '../../src/utils/ast-utils';
+import { addUpdateTask } from '../../src/utils/update-task';
 
-const updateAngularCLI = externalSchematic('@schematics/update', 'update', {
-  packages: ['@angular/cli'],
-  from: '7.1.0',
-  to: '7.2.2',
-  force: true
-});
+const updateAngularCLI = addUpdateTask('@angular/cli', '7.2.2');
 
 const updateTypescript = updateJsonInTree('package.json', json => {
   json.devDependencies = json.devDependencies || {};

--- a/packages/schematics/migrations/update-7-6-0/update-7-6-0.ts
+++ b/packages/schematics/migrations/update-7-6-0/update-7-6-0.ts
@@ -1,10 +1,4 @@
-import {
-  Rule,
-  chain,
-  externalSchematic,
-  SchematicContext,
-  Tree
-} from '@angular-devkit/schematics';
+import { chain, Rule, Tree } from '@angular-devkit/schematics';
 
 import { ReplaceChange } from '@schematics/angular/utility/change';
 import { getSourceNodes } from '@schematics/angular/utility/ast-utils';
@@ -14,8 +8,10 @@ import * as ts from 'typescript';
 import {
   updateJsonInTree,
   readJsonInTree,
-  insert
+  insert,
+  addDepsToPackageJson
 } from '../../src/utils/ast-utils';
+import { addUpdateTask } from '../../src/utils/update-task';
 import { formatFiles } from '../../src/utils/rules/format-files';
 
 const addExtensionRecommendations = updateJsonInTree(
@@ -404,21 +400,13 @@ const setDefaults = updateJsonInTree('angular.json', json => {
 });
 
 const updateAngularCLI = chain([
-  externalSchematic('@schematics/update', 'update', {
-    packages: ['@angular/cli'],
-    from: '7.2.2',
-    to: '7.3.1',
-    force: true
-  }),
-  updateJsonInTree('package.json', json => {
-    json.devDependencies = json.devDependencies || {};
-    json.devDependencies = {
-      ...json.devDependencies,
-      '@angular/cli': '7.3.1',
+  addUpdateTask('@angular/cli', '7.3.1'),
+  addDepsToPackageJson(
+    {},
+    {
       '@angular-devkit/build-angular': '~0.13.1'
-    };
-    return json;
-  })
+    }
+  )
 ]);
 
 export default function(): Rule {

--- a/packages/schematics/src/utils/update-task.ts
+++ b/packages/schematics/src/utils/update-task.ts
@@ -1,0 +1,99 @@
+import {
+  chain,
+  Rule,
+  SchematicContext,
+  TaskConfiguration,
+  TaskConfigurationGenerator,
+  TaskExecutorFactory,
+  Tree
+} from '@angular-devkit/schematics';
+import { platform } from 'os';
+import { Observable } from 'rxjs';
+import { spawn } from 'child_process';
+
+let taskRegistered = false;
+
+export function addUpdateTask(pkg: string, to: string): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    // Workflow should always be there during ng update but not during tests.
+    if (!context.engine.workflow) {
+      return;
+    }
+    if (!taskRegistered) {
+      const engineHost = (context.engine.workflow as any)._engineHost;
+      engineHost.registerTaskExecutor(createRunUpdateTask());
+
+      taskRegistered = true;
+    }
+    (context.engine as any)._taskSchedulers.forEach(scheduler => {
+      if (
+        scheduler._queue.peek() &&
+        scheduler._queue.peek().configuration.name === 'RunUpdate' &&
+        scheduler._queue.peek().configuration.options.package === pkg
+      ) {
+        scheduler._queue.pop();
+      }
+    });
+
+    context.addTask(new RunUpdateTask(pkg, to));
+  };
+}
+
+interface UpdateTaskOptions {
+  package: string;
+  to: string;
+}
+
+class RunUpdateTask implements TaskConfigurationGenerator<UpdateTaskOptions> {
+  constructor(private _pkg: string, private _to: string) {}
+
+  toConfiguration(): TaskConfiguration<UpdateTaskOptions> {
+    return {
+      name: 'RunUpdate',
+      options: {
+        package: this._pkg,
+        to: this._to
+      }
+    };
+  }
+}
+
+function createRunUpdateTask(): TaskExecutorFactory<UpdateTaskOptions> {
+  return {
+    name: 'RunUpdate',
+    create: () => {
+      return Promise.resolve(
+        (options: UpdateTaskOptions, context: SchematicContext) => {
+          context.logger.info(`Updating ${options.package} to ${options.to}`);
+          const spawnOptions = {
+            stdio: [process.stdin, process.stdout, process.stderr],
+            shell: true
+          };
+          const ng =
+            platform() === 'win32'
+              ? '.\\node_modules\\.bin\\ng'
+              : './node_modules/.bin/ng';
+          const args = [
+            'update',
+            `${options.package}@${options.to}`,
+            '--force',
+            '--allow-dirty'
+          ].filter(e => !!e);
+          return new Observable(obs => {
+            spawn(ng, args, spawnOptions).on('close', (code: number) => {
+              if (code === 0) {
+                obs.next();
+                obs.complete();
+              } else {
+                const message = `${
+                  options.package
+                } migration failed, see above.`;
+                obs.error(new Error(message));
+              }
+            });
+          });
+        }
+      );
+    }
+  };
+}

